### PR TITLE
add type annotations to torch.nn.modules.normalization

### LINKED ISSUE #49034: Enable torch.nn.modules.normalization typechecks during CI
`TestTypeHints.test_run_mypy` should pass with `[mypy-torch.nn.modules.normalization]` deleted from mypy.ini

cc @ezyang @malfet @rgommers @xuzhao9 @gramster

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Fixes #49034

### CODE REVIEW

File: torch/nn/modules/normalization.py
Comment (rgommers) [inline torch/nn/modules/normalization.py:144]: This should be `Tuple[int, ...]`, because there's an explicit coercion on line 153 and the tuple is of variable length.

File: torch/nn/modules/normalization.py
Comment (rgommers) [inline torch/nn/modules/normalization.py:10]: Related to comment further down, replace with `Tuple`

File: torch/nn/modules/normalization.py
Comment (rgommers) [reply torch/nn/modules/normalization.py:144]: Also, `_shape_t` should be changed from `List[int]` to `Sequence[int]` on line 83 for consistency it looks like.

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks @guilhermeleobas. Just a couple of minor comments.

**Review by rgommers — APPROVED** [review]
LGTM now, thanks @guilhermeleobas

File: torch/nn/modules/normalization.py
Comment (walterddr) [inline torch/nn/modules/normalization.py:173]: wonder why it was not caught before.

File: torch/nn/modules/normalization.py
Comment (walterddr) [inline torch/nn/modules/normalization.py:83]: do we have a test to validate JIT-ability? since we knew Sequence & Tuple doesnt work on JIT.

**Comment by guilhermeleobas** [comment]
Do not merge this PR until one checks if the annotations introduce any regression. See:
https://github.com/pytorch/pytorch/pull/49564#issuecomment-752916072

File: torch/nn/modules/normalization.py
Comment (guilhermeleobas) [reply torch/nn/modules/normalization.py:83]: Actually this is not necessary. I am reverting this particular change

**Comment by guilhermeleobas** [comment]
Tests are covered by test cases in the `TestJitGeneratedModule`. See:
```
test/test_jit.py::TestJitGeneratedModule::test_nn_LayerNorm_1d_elementwise_affine
test/test_jit.py::TestJitGeneratedModule::test_nn_LayerNorm_1d_empty_elementwise_affine
test/test_jit.py::TestJitGeneratedModule::test_nn_LayerNorm_1d_no_elementwise_affine
test/test_jit.py::TestJitGeneratedModule::test_nn_LayerNorm_3d_elementwise_affine
test/test_jit.py::TestJitGeneratedModule::test_nn_LayerNorm_3d_no_elementwise_affine
```

File: torch/nn/modules/normalization.py
Comment (walterddr) [inline torch/nn/modules/normalization.py:153]: normalized_
```suggestion
            normalized_shape_ = (normalized_shape,)
        self.normalized_shape = tuple(normalized_shape_)
```

Comment (walterddr) [reply torch/nn/modules/normalization.py:153]: also wondering do we even need the second `tuple()` wrapper?

**Review by walterddr — COMMENTED** [review]
looks good. one minor comment. 

also could you confirm the coverage for LayerNorm1d constructor arguments in `torch.test._internal.common_nn` actually consist of non-tuple type input? from what I see all of those constructor_args are lists. not sure if they collapse into single integer

File: torch/nn/modules/normalization.py
Comment (guilhermeleobas) [reply torch/nn/modules/normalization.py:153]: I guess the second `tuple()` is to make sure `self.normalized_shape` is a tuple and not a list, for instance.

Comment (guilhermeleobas) [reply torch/nn/modules/normalization.py:153]: Also, I do not think the suggested change works as expected. If the input arg `normalized_shape` is a list, then the if block will never be executed and `normalized_shape_` will not be defined.

**Review by malfet — APPROVED** [review]

**Review by walterddr — APPROVED** [review]

**Pull request closed without merging**
